python/test/test_meas.py

```diff
@@ -200,17 +200,36 @@ def test_gaussian():
     input_space = dp.vector_domain(dp.atom_domain(T=float)), dp.l2_distance(T=float)
     (input_space >> dp.m.then_gaussian(1.))([1., 2., 3.])
 
+
 def test_report_noisy_max_gumbel():
     input_domain = dp.vector_domain(dp.atom_domain(T=dp.usize))
 
     input_metric = dp.linf_distance(T=dp.usize)
     meas = (input_domain, input_metric) >> dp.m.then_report_noisy_max_gumbel(1., "max")
-    print(meas(list(range(10))))
+    # fails with very small probability
+    assert meas([0, 0, 20, 0]) == 2  # because score 2 is by far the greatest
     assert meas.map(2) == 4
 
     input_metric = dp.linf_distance(monotonic=True, T=dp.usize)
     meas = (input_domain, input_metric) >> dp.m.then_report_noisy_max_gumbel(1., "max")
-    print(meas(list(range(10))))
+    # fails with very small probability
+    assert meas([0, 0, 20, 0]) == 2  # because score 2 is by far the greatest
+    assert meas.map(2) == 2
+
+
+def test_report_noisy_max_exponential():
```

Comment:
Not sure where this should live, but I think it would be great to have somewhere a benchmark showing where it makes sense to use either one of them to discuss maybe somewhere the tradeoff between the two, so that people reading the source code get some intuition why there are 2 mechanisms for argmax currently.

Reply:
There isn't really a good reason yet until bounded-range is added, maybe we can do a pass then.